include/constantine/protocols/ethereum_verkle_ipa.h

```diff
@@ -0,0 +1,228 @@
+/** Constantine
+ *  Copyright (c) 2018-2019    Status Research & Development GmbH
+ *  Copyright (c) 2020-Present Mamy André-Ratsimbazafy
+ *  Licensed and distributed under either of
+ *    * MIT license (license terms in the root directory or at http://opensource.org/licenses/MIT).
+ *    * Apache v2 license (license terms in the root directory or at http://www.apache.org/licenses/LICENSE-2.0).
+ *  at your option. This file may not be copied, modified, or distributed except according to those terms.
+ */
+#ifndef __CTT_H_ETHEREUM_VERKLE_IPA__
+#define __CTT_H_ETHEREUM_VERKLE_IPA__
+
+#include "constantine/core/datatypes.h"
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum __attribute__((__packed__)) {
+    cttEthVerkleIpa_Success,
+    cttEthVerkleIpa_VerificationFailure,
+    cttEthVerkleIpa_InputsLengthsMismatch,
+    cttEthVerkleIpa_ScalarZero,
+    cttEthVerkleIpa_ScalarLargerThanCurveOrder,
+    cttEthVerkleIpa_EccInvalidEncoding,
+    cttEthVerkleIpa_EccCoordinateGreaterThanOrEqualModulus,
+    cttEthVerkleIpa_EccPointNotOnCurve,
+    cttEthVerkleIpa_EccPointNotInSubGroup
+} ctt_eth_verkle_ipa_status;
+
+static const char* ctt_eth_verkle_ipa_status_to_string(ctt_eth_verkle_ipa_status status) {
+  static const char* const statuses[] = {
+    "cttEthVerkleIpa_Success",
+    "cttEthVerkleIpa_VerificationFailure",
+    "cttEthVerkleIpa_InputsLengthsMismatch",
+    "cttEthVerkleIpa_ScalarZero",
+    "cttEthVerkleIpa_ScalarLargerThanCurveOrder",
+    "cttEthVerkleIpa_EccInvalidEncoding",
+    "cttEthVerkleIpa_EccCoordinateGreaterThanOrEqualModulus",
+    "cttEthVerkleIpa_EccPointNotOnCurve",
+    "cttEthVerkleIpa_EccPointNotInSubGroup"
+  };
+  size_t length = sizeof statuses / sizeof *statuses;
+  if (0 <= status && status < length) {
+    return statuses[status];
+  }
+  return "cttEthVerkleIpa_InvalidStatusCode";
+}
+
+// Opaque types for Nim-defined types
+typedef struct Fr Fr;
+typedef struct Banderwagon Banderwagon;
+typedef struct EC_TwEdw EC_TwEdw;
+typedef struct Fp Fp;
+
+typedef union {
+    uint8_t u8;
+    uint16_t u16;
+    uint32_t u32;
+    uint64_t u64;
+    unsigned int u;
+} SomeUnsignedInt;
+
+typedef struct {
+    SomeUnsignedInt* values;  
+    size_t length;
+} SomeUnsignedInt_Array;
+
+// Define types for openArray
+typedef struct {
+    Fr* data;    
+    size_t len;   
+} Fr_BanderWagon_OpenArray;
+
+typedef struct {
+    EC_TwEdw* data;    
+    size_t len;         
+} EC_TwEdw_Fp_Banderwagon_OpenArray;
+
+typedef struct {
+    byte value[32];  // 32-byte array for field element
+} Fp_Banderwagon;
```

Comment:
should be autogenerated in `constantine/include/constantine/curves/banderwagon.h`